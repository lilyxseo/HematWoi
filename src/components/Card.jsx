export default function Card({ title, actions, children }) {
  return (
    <div className="card">
      {(title || actions) && (
        <div className="flex items-center justify-between mb-2">
          {title && <h2 className="font-semibold">{title}</h2>}
          {actions}
        </div>
      )}
      {children}
    </div>
  );
}
