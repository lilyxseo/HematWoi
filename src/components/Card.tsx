import { PropsWithChildren } from 'react';
import clsx from 'clsx';

export default function Card({ children, className }: PropsWithChildren<{ className?: string }>) {
  return <div className={clsx('bg-white dark:bg-gray-800 rounded-2xl shadow p-4', className)}>{children}</div>;
}
