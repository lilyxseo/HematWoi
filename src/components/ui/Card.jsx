import React from "react";
export function Card({ className = "", children }) {
  return (
    <section
      className={`card ${className}`}
      style={{ containerType: "inline-size", containerName: "card" }}
    >
      {children}
    </section>
  );
}
export function CardHeader({ title, subtitle, extra }) {
  return (
    <header
      style={{
        display: "flex",
        alignItems: "center",
        justifyContent: "space-between",
        gap: "12px",
        flexWrap: "wrap",
      }}
    >
      <div>
        {title && <div className="card-title">{title}</div>}
        {subtitle && <div className="card-sub">{subtitle}</div>}
      </div>
      {extra && (
        <div
          className="actions"
          style={{ display: "flex", gap: "8px", flexWrap: "wrap" }}
        >
          {extra}
        </div>
      )}
    </header>
  );
}
export function CardBody({ children }) {
  return <div>{children}</div>;
}
export function CardFooter({ children }) {
  return <footer style={{ marginTop: "auto" }}>{children}</footer>;
}
export default { Card, CardHeader, CardBody, CardFooter };
