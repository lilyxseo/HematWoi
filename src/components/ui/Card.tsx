import { HTMLAttributes } from 'react';
import { cn } from '../../lib/utils';

export const Card = ({ className, ...props }: HTMLAttributes<HTMLDivElement>) => {
  return (
    <div
      className={cn(
        'rounded-2xl border border-slate-200 bg-white p-4 shadow-sm md:p-6',
        className
      )}
      {...props}
    />
  );
};
