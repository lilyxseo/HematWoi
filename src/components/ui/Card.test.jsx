/* @vitest-environment jsdom */
import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import { Card, CardHeader, CardBody } from './Card';

describe('Card component', () => {
  it('renders structure', () => {
    const { asFragment, getByText } = render(
      <Card>
        <CardHeader title="Title" subtitle="Sub" extra={<button>Act</button>} />
        <CardBody>Content</CardBody>
      </Card>
    );
    expect(getByText('Title')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
