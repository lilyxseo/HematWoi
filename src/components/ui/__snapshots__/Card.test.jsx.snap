// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card component > renders structure 1`] = `
<DocumentFragment>
  <section
    class="card "
    style="container-type: inline-size; container-name: card;"
  >
    <header
      style="display: flex; align-items: center; justify-content: space-between; gap: 12px; flex-wrap: wrap;"
    >
      <div>
        <div
          class="card-title"
        >
          Title
        </div>
        <div
          class="card-sub"
        >
          Sub
        </div>
      </div>
      <div
        class="actions"
        style="display: flex; gap: 8px; flex-wrap: wrap;"
      >
        <button>
          Act
        </button>
      </div>
    </header>
    <div>
      Content
    </div>
  </section>
</DocumentFragment>
`;
